plugins/inputs/mongodb/mongodb.go

```diff
@@ -135,20 +135,24 @@ func (m *MongoDB) Init() error {
 			opts.ReadPreference = readpref.Nearest()
 		}
 
+		isReachable := true
 		client, err := mongo.Connect(ctx, opts)
 		if err != nil {
-			return fmt.Errorf("unable to connect to MongoDB: %q", err)
+			m.Log.Errorf("unable to create MongoDB client: %q", err)
```

Comment:
I think we still want to return here in case of an error, otherwise `client.Ping` will get called and this could lead to a panic because `client` could be nil if there was an error.

Reply:
I think this is what causes the behavior I wanted to avoid with this PR. If you return an error in an `init`, Telegraf exits.